src/gnc_environment.cpp

```diff
@@ -0,0 +1,61 @@
+//
+// src/gnc_environment.hpp
+// PSim
+//
+// Contributors:
+//   Kyle Krol  kpk63@cornell.edu
+//
+// Pathfinder for Autonomous Navigation
+// Space Systems Design Studio
+// Cornell Univeristy
+//
+
+#include <gnc_constants.hpp>
+#include <gnc_environment.hpp>
+#include <gnc_utilities.hpp>
+
+#include <cmath>
+
+namespace gnc {
+namespace env {
+
+void earth_attitude(double t, lin::Vector4d &q) {
+  using namespace lin;
+  // Determine a transformation from ecef0 to ecef0p
+  Vector4d q_ecef0p_ecef0({
```

Comment:
What's the difference between ECEF0P and ECEF0?

Reply:
It's sort of a long story that @nhz2 can explain better. But essentially you're treating the Earth's precession and rotation independently from each other and therefore get these intermediate frames.

So you go ECI -> ECEF0 (ECEF at time 0) -> ECEF0p (no precession just z axis rotation) -> ECEF (precession and z axis rotation) or something along those lines.